### ISSUE TITLE
standard navigation keys do not work in shell.

### ISSUE THREAD

**Issue by dan-da** [maidsafe/safe-api#335]
For example arrow keys, home/end do not work as expected and just print garbage control chars.

I opened [an issue](https://github.com/phsym/shrust/issues/7) for this in shrust, so this is just for tracking it here.

In following example, I pressed *home* key from end of line, then *left arrow*.
```
> dog safe://hnyynyiptjhhhzsicasjqhrdnmgu78bm4bcrj8o7q6qazai8sfj3kbt8nqbnc^[[H^[[D
```

**Comment by dan-da** [comment]
Ok, I did a little research and found [linefeed](https://github.com/murarth/linefeed) which is quite nice as it supports navigation, tab completion, history, etc.   Basically, running the included demo, everything just worked as I expected.  I looked for others, but didn't find anything else that really seemed to fit the bill.

I don't know what evaluation/criteria went into choosing shrust, but maybe we should think about moving to linefeed instead?   For me, the present shell with shrust is not really usable, and the linefeed demo worked just right and is slick.

@bochaco, if you approve of the idea, I would be willing take on such an effort, or assist as needed.

**Comment by bochaco** [comment]
Yes, and just FYI, I noticed shrust works a bit better on windows, with arrow keys at least. We used shrust just because it was extremely simple to come up with first version of the interactive shell.
Now, as requirements if you will, what we need is to be able to keep an state across commands, e.g. the Safe and safe-authd connections, plus being able to integrate it with structopts so we re-use the validations and commands operations completely from normal cli (there is only one exeception which is or the subscribe command, but others commands call the same logic used by normal cli: https://github.com/maidsafe/safe-api/blob/master/safe-cli/shell.rs#L142).
Thus, if that's all possible with linefeed and you think it'll give us a better interactive shell, then there shouldn't be anything against your idea @dan-da , for sure it's worthwhile to give it try.

**Comment by dan-da** [comment]
Ok, yeah I will play with it and see if it is easy to drop-in or there are gotchas.

incidentally, I noticed that ctrl-u works in the shrust shell at least.  :+1:
